src/controllers/comment.controller.js

```diff
@@ -0,0 +1,124 @@
+import CommentService from '../services/comment.service';
+import responseHandlers from '../utils/responseHandlers';
+import statusCodes from '../utils/statusCodes';
+import customMessages from '../utils/customMessages';
+import RequestService from '../services/request.service';
+
+const { saveNewComment, updateComment, deleteComment } = CommentService;
+const { successResponse, errorResponse } = responseHandlers;
+const { serverError, created, notFound, ok, badRequest } = statusCodes;
+const {
+    commentAdded,
+    noRequestYet,
+    requestAndCommentsRetrieved,
+    requestButNoCommentYet,
+    commentUpdatedSuccess,
+    commentDeleted,
+    requestNotExists,
+    noCommentYet,
+    commentsRetrieved
+} = customMessages;
+const { getAllRequestsOfAUser, getRequestFromDb } = RequestService;
+/**
+ * @description class CommentController handles all comments controllers methods
+ */
+export default class CommentController {
+    /**
+     * @param {object} req
+     * @param {object} res
+     * @returns {object} response to user
+     * @description it will call comment service to save a new comment in db
+     */
+    static addNewComment = async (req, res) => {
+        const { id } = req.sessionUser;
+        const { requestId, comment } = req.body;
+        const savedComment = await saveNewComment({ userId: id, requestId, comment });
+        successResponse(res, created, commentAdded, null, savedComment);
+    }
+
+    /**
+     * @param {object} req
+     * @param {object} res
+     * @returns {object} response to user
+     * @description it responds to user about the update result
+     */
+    static updateComments = async (req, res) => {
+        const { commentId } = req.params;
+        const { requestId, comment } = req.body;
+        const commentData = { id: commentId, requestId, comment };
+        const updatedCommentToReturn = await updateComment(commentData);
+        successResponse(res, ok, commentUpdatedSuccess, null, updatedCommentToReturn);
+    }
+
+    /**
+     * @param {object} req
+     * @param {object} res
+     * @returns {object} sends response to the user
+     * @description it sends list of all requests and their comments to user
+     */
+    static getUserRequestsAndComments = async (req, res) => {
+        const currUserRole = req.sessionUser.role;
+        const currUserId = req.sessionUser.id;
+        let requestsToShow;
+        let counter = 0;
+        if (currUserRole === 'manager') {
+            requestsToShow = await getAllRequestsOfAUser();
```

Comment:
This function is confusing, does it `get all requests of a user?` if yes where is the id of the user? I can see it's the same function you used below here, can you rename this function to something that is more descriptive to what it does?

Reply:
@Muhire-Josue I changed its name to `getAllRequestsFromDb` Actually this method is from `request.service`, it returns list of requests from the database if it is passed userId; otherwise, it returns all of the requests, you can read its description in `request.service`,